contrib/gazebo/gazebo_sawyer_env.py

```diff
@@ -0,0 +1,499 @@
+        self.reward_type = reward_type
+        self._tip_name = end_point
+        self._limb_name = limb
+        self._limb = intera_interface.Limb(limb)
+        self._gripper = intera_interface.Gripper()
+        self._control_mode = control_mode
+        # get the joint limits
+        self._joint_limits = rospy.wait_for_message(
+            '/robot/joint_limits', JointLimits, timeout=5)
+        # set the action space
+        self._set_action_space()
+        self.target_range = target_range
+        self.target_in_the_air = target_in_the_air
+        self.obj_range = obj_range
+        self.start_angles = {
+            'right_j0': initial_qpos['right_j0'],
+            'right_j1': initial_qpos['right_j1'],
+            'right_j2': initial_qpos['right_j2'],
+            'right_j3': initial_qpos['right_j3'],
+            'right_j4': initial_qpos['right_j4'],
+            'right_j5': initial_qpos['right_j5'],
+            'right_j6': initial_qpos['right_j6']
+        }
+        self.model_states = rospy.wait_for_message(
+            "/gazebo/model_states", ModelStates, timeout=5)
+        rospy.Subscriber('/gazebo/model_states', ModelStates,
+                         self.update_model_states)
+        GazeboEnv.__init__(self, initial_qpos=initial_qpos)
+        self._observation_space = Box(
+            -np.inf, np.inf, shape=self._get_obs()['observation'].shape)
+
+    def _set_action_space(self):
+        # For sawyer's joints
+        # [head_pan, right_gripper_l_finger_joint, right_gripper_r_finger_joint, right_j0, right_j1,
+        # right_j2, right_j3, right_j4, right_j5, right_j6]
+        # For our use
+        # joints[3:10]
+        # set the action space depending on different control modes
+        if self._control_mode == 'position':
+            lower_bounds = np.array(self._joint_limits.position_lower[3:10])
+            upper_bounds = np.array(self._joint_limits.position_upper[3:10])
+        elif self._control_mode == 'velocity':
+            lower_bounds = np.zeros(7)
+            upper_bounds = np.array(self._joint_limits.velocity[3:10])
+        elif self._control_mode == 'effort':
+            lower_bounds = np.zeros(7)
+            upper_bounds = np.array(self._joint_limits.effort[3:10])
+        else:
+            # should raise an unknown mode exception
+            raise ControlModeUnknownErr(
+                "Control mode %s is not known!" % self._control_mode)
+
+        self._action_space = Box(
+            np.concatenate((lower_bounds, np.array([0]))),
+            np.concatenate((upper_bounds, np.array([100]))))
+
+    def update_model_states(self, data):
+        self.model_states = data
+
+    def compute_reward(self, achieved_goal, goal, info=None):
+        # Compute distance between goal and the achieved goal.
+        d = goal_distance(achieved_goal, goal)
+        if self.reward_type == 'sparse':
+            return -(d > self.distance_threshold).astype(np.float32)
```

Comment:
@eric-heiden and I found between 10 and 100 to be sufficient for TRPO, IIRC.

You can make the completion bonus for sparse rewards the same. As the rewards are sparse, the relative magnitude is (mostly) irrelevant, but it is convenient to make it the same as the dense reward completion bonus, so that that magnitude of other types of rewards (e.g. entropy) is approximately the same between them.